api/provider/aws/generic_load_balancer.go

```diff
@@ -0,0 +1,33 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/service/elb"
+	alb "github.com/aws/aws-sdk-go/service/elbv2"
+)
+
+type genericLoadBalancer struct {
```

Comment:
Could this object be simplified? Seems like you could just use typecasting in `GetScheme()`

Reply:
Not sure what you mean? Can you please provide an example?

Reply:
My typecasting statement was written in haste, although i'm sure there is a way you could take that approach here as well.

Anyway, it seems redundant to have bools for `isELB` or `isALB` when you could just return an ELB or ALB as part of `describeLoadBalancer`. Then in `GetScheme()`, instead of checking `isELB` or `isALB`, you could just check for `nil`.

Reply:
Oh right - yes you're correct. I was in fact doing exactly what you described. But the calling code which looked checked for ELB or an ALB looked cleaner with the bools:

```go
loadBalancer := DescribeLoadBalancer(loadBalancerID)

// without the bools
if loadbalancer.ELB != nil { ... }

// but having isELB & isALB property, it looks more like (which seemed subjectively cleaner)
if loadbalancer.isELB { ... }
```